internal/services/watchlist/watchlist.go

```diff
@@ -0,0 +1,62 @@
+package watchlist
+
+import (
+	"Tradeasy/config"
+	"Tradeasy/internal/model"
+	"errors"
+)
+
+func CreateWatchlist(CrReq CreateRequest) (CrRes CreateResponse, err error) {
+	var wl model.Watchlist
+	wl.Name = CrReq.WatchlistName
+	er := config.DB.Table("watchlist").Where("name = ?", CrReq.WatchlistName).First(&wl).Error
+	if er == nil {
+		return CrRes, errors.New("watchlist name already exists")
+	}
+	err = config.DB.Table("watchlist").Create(&wl).Error
+	if err != nil {
+		return CrRes, errors.New("failed to create watchlist")
+	}
+	CrRes.Message = "Watchlist created"
+	CrRes.WatchlistId = wl.Id
+
+	return CrRes, nil
+}
+
+func AddStockEntry(AddReq AddStockRequest, watchlistId int) (AddRes AddStockResponse, err error) {
+	var uwl model.UserWatchlist
+	uwl.Userid = AddReq.UserId
```

Comment:
Why model data is getting filled without checking if request is valid or not?

Reply:
done